packages/node_modules/@ciscospark/plugin-messages/src/messages.js

```diff
@@ -57,65 +55,105 @@ const Messages = SparkPlugin.extend({
+        const deletedEvent = this.getMessageEvent(activity, SDK_EVENT.TO_APP.EVENT_TYPE.DELETED);
+
+        if (deletedEvent) {
+          debug(`messages "deleted" payload: \
+            ${JSON.stringify(deletedEvent)}`);
+          this.trigger(SDK_EVENT.TO_APP.EVENT_TYPE.DELETED, deletedEvent);
+        }
         break;
+
       default: {
         break;
       }
     }
   },
 
   /**
-   * Constructs an event data object for the "created" event,
-   * adhering to Hydra's Message details.
-   * @see https://developer.webex.com/docs/api/v1/messages/get-message-details
+   * Constructs the data object for an event on the messages resource,
+   * adhering to Hydra's Webehook data structure messages.
    * @param {Object} activity from mercury
+   * @param {Object} event type of "webhook" event
    * @returns {Object} constructed event
    */
-  createMessagesEventData(activity) {
-    const roomType =
-      activity.target.tags.includes('ONE_ON_ONE') ? 'direct' : 'group';
-
-    const event = {
-      id: constructHydraId(hydraTypes.MESSAGE, activity.id),
-      roomId: constructHydraId(hydraTypes.ROOM, activity.target.id),
-      roomType,
-      text: activity.object.displayName,
-      personId: constructHydraId(hydraTypes.PEOPLE, activity.actor.id),
-      personEmail: activity.actor.emailAddress || activity.actor.entryEmail,
-      created: activity.published
-    };
+  getMessageEvent(activity, event) {
+    try {
+      const sdkEvent = cloneDeep(this.eventEnvelope);
+
+      sdkEvent.event = event;
+      sdkEvent.data.created = activity.published;
+      sdkEvent.actorId = constructHydraId(hydraTypes.PEOPLE, activity.actor.id);
+      sdkEvent.data.roomType =
+        activity.target.tags.includes(SDK_EVENT.FROM_WEBEX.ACTIVITY_TAG.ONE_ON_ONE) ?
+          SDK_EVENT.TO_APP.SPACE_TYPE.DIRECT :
+          SDK_EVENT.TO_APP.SPACE_TYPE.GROUP;
+      sdkEvent.data.roomId =
+        constructHydraId(hydraTypes.ROOM, activity.target.id);
+      sdkEvent.data.personId =
+        constructHydraId(hydraTypes.PEOPLE, activity.actor.id);
+      sdkEvent.data.personEmail =
+        activity.actor.emailAddress || activity.actor.entryEmail;
+
+      if (event !== SDK_EVENT.TO_APP.EVENT_TYPE.DELETED) {
+        const files = getHydraFiles(activity);
 
-    const files = getHydraFiles(activity);
+        sdkEvent.data.id = constructHydraId(hydraTypes.MESSAGE, activity.id);
+        sdkEvent.data.text = activity.object.displayName;
+        if (files.length) {
+          sdkEvent.data.files = files;
+        }
+      }
+      else {
+        sdkEvent.data.id =
+          constructHydraId(hydraTypes.MESSAGE, activity.object.id);
+      }
 
-    if (files.length) {
-      event.files = files;
+      return sdkEvent;
     }
+    catch (e) {
+      console.error(`Unable to generate SDK event from mercury \
+'socket activity for message:${event} event: ${e.message}`);
 
-    return event;
+      return null;
```

Comment:
What does the calling function do with the `null`? Is it better to have a `null` then just handle a thrown error? What is the advantage of `null`?

Reply:
In the switch/case statement in the function directly above, the calling function checks if this function returns null.  If so it does not call the registered callback since we don't have a good event.